File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/10448_유레카이론/10448.cpp
#include <stdio.h>
#include <vector>
using namespace std;

int T, K;
int is_eureka[1001];
vector<int> triangle_nums;

int main()
{
    for (int i = 1; ; i++)
    {
        int num = i * (i + 1) / 2;
        if (num > 1000)
            break;
        triangle_nums.push_back(num); 
    }

    for (int i = 0; i < triangle_nums.size(); i++)
    {
        for (int j = i; j < triangle_nums.size(); j++)
        {
            for (int k = j; k < triangle_nums.size(); k++)
            {
                int num = triangle_nums[i] + triangle_nums[j] + triangle_nums[k];
                if (num > 1000)
                    break;
                is_eureka[num] = 1;
            }
        }
    }

    scanf("%d", &T);
    while (T--)
    {
        scanf("%d", &K);
        printf("%d\n", is_eureka[K]);
    }
    return 0;
}

File: Part1_강의자료/Ch04_전체탐색_시뮬레이션/문제별코드/10448_유레카이론/10448.py
triangle_nums = []
is_eureka = [0] * 1001

for i in range(1, 1001):
    num = i * (i + 1) // 2
    if num > 1000:
        break
    triangle_nums.append(num)

for i in range(len(triangle_nums)):
    for j in range(i, len(triangle_nums)):
        for k in range(j, len(triangle_nums)):
            num = triangle_nums[i] + triangle_nums[j] + triangle_nums[k]
            if num > 1000:
                break
            is_eureka[num] = 1

T = int(input())
for _ in range(T):
    K = int(input())
    print(is_eureka[K])
